Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/SearchRetry/OneWayRetry.hs

```diff
@@ -0,0 +1,118 @@
+    oneWaySearchRetry,
+  )
+where
+
+import qualified Beckn.Types.Core.Taxi.Select as Select
+import qualified Domain.Action.UI.Search.Common as DSearch
+import qualified Domain.Action.UI.Search.OneWay as AUSO
+import qualified Domain.Types.Merchant as Merchant
+import qualified Domain.Types.Person as Person
+import qualified Domain.Types.Person.PersonFlowStatus as DPFS
+import Domain.Types.SearchRequest (SearchRequest)
+import qualified Domain.Types.SearchRetry as DSearchRetry
+import Kernel.External.Maps.Types
+import Kernel.Prelude
+import qualified Kernel.Storage.Esqueleto as DB
+import Kernel.Storage.Hedis (HedisFlow)
+import Kernel.Types.Common hiding (id)
+import Kernel.Types.Error (SearchRequestError (SearchRequestNotFound))
+import Kernel.Types.Id
+import Kernel.Utils.Common
+import Storage.CachedQueries.CacheConfig
+import qualified Storage.Queries.Person.PersonFlowStatus as QPFS
+import qualified Storage.Queries.SearchRequest as QSearchRequest
+import qualified Storage.Queries.SearchRetry as QSearchRetry
+import Tools.Metrics
+import qualified Tools.Metrics as Metrics
+
+newtype SearchRetryReq = SearchRetryReq
+  { retryType :: Select.RetryType
+  }
+  deriving (Generic, FromJSON, ToJSON, Show, ToSchema)
+
+oneWaySearchRetry ::
+  ( HasCacheConfig r,
+    EncFlow m r,
+    HasFlowEnv m r '["searchRequestExpiry" ::: Maybe Seconds],
+    HedisFlow m r,
+    EsqDBFlow m r,
+    HedisFlow m r,
+    CoreMetrics m,
+    HasBAPMetrics m r
+  ) =>
+  Person.Person ->
+  Merchant.Merchant ->
+  SearchRetryReq ->
+  Id SearchRequest ->
+  m AUSO.OneWaySearchRes
+oneWaySearchRetry person merchant req parentSearchId = do
+  searchRequestRow <- QSearchRequest.findById parentSearchId >>= fromMaybeM (SearchRequestNotFound parentSearchId.getId)
+  let destination =
+        case searchRequestRow.toLocation of
+          Just toLocation ->
+            Just
+              DSearch.SearchReqLocation
+                { gps =
+                    LatLong
+                      { lat = toLocation.lat,
+                        lon = toLocation.lon
+                      },
+                  address = toLocation.address
+                }
+          Nothing -> Nothing
+  let justDestination = fromMaybe (error "Destination is Nothing") destination
+  let origin =
+        DSearch.SearchReqLocation
+          { gps =
+              LatLong
+                { lat = searchRequestRow.fromLocation.lat,
+                  lon = searchRequestRow.fromLocation.lon
+                },
+            address = searchRequestRow.fromLocation.address
+          }
+  fromLocation <- DSearch.buildSearchReqLoc origin
+  toLocation <- mapM DSearch.buildSearchReqLoc destination
+  now <- getCurrentTime
+  searchRequest <- DSearch.buildSearchRequest person fromLocation toLocation searchRequestRow.maxDistance searchRequestRow.distance now searchRequestRow.bundleVersion searchRequestRow.clientVersion searchRequestRow.device searchRequestRow.estimatedRideDuration
```

Comment:
seems like `searchRequest` is the same as `searchRequestRow`, only `id` is different. Why do we need to store it?

Reply:
done